dockerclient.go

```diff
@@ -97,6 +97,41 @@ func (client *DockerClient) doRequest(method string, path string, body []byte, h
 	return data, nil
 }
 
+func (client *DockerClient) doStreamRequest(method string, path string, in io.Reader, headers map[string]string) (io.ReadCloser, error) {
```

Comment:
What's the difference between this and `doRequest`? Can the two be refactored so they share most of the same code?